### PR TITLE
Share Image activity

### PR DESCRIPTION
**Pull request by Subhankar29**
Fixes issue #543 

Changes: Made a new layout for the Share Image activity. Added Edit text to share text with the Image.
Currently, the following features are working:

- [x] Share through share intent.
- [x] Share to Facebook.
- [x] Share Directly to Instagram.
- [ ] Share to Twitter.

Screenshots for the change: 
screenshot_2017-06-11-19-49-52-613

### CODE REVIEW

File: app/src/main/java/vn/mbm/phimp/me/editor/editimage/EditImageActivity.java [hunk 1]
```diff
@@ -224,6 +234,21 @@ public void onFling(MotionEvent e1, MotionEvent e2, float velocityX, float veloc
         });
     }
 
+    private void shareImage() {
+        if(mOpTimes>0) {
```
Comment (pa1pal) [inline app/src/main/java/vn/mbm/phimp/me/editor/editimage/EditImageActivity.java]: Why this check? It force user to edit image at least once to enable the share button. User can share any image edited or not.

File: app/src/main/java/vn/mbm/phimp/me/editor/editimage/EditImageActivity.java [hunk 2]
```diff
@@ -224,6 +234,21 @@ public void onFling(MotionEvent e1, MotionEvent e2, float velocityX, float veloc
...
+            shareIntent.putExtra(FILE_PATH, filePath);
+            shareIntent.putExtra(EXTRA_OUTPUT, saveFilePath);
+            shareIntent.putExtra(IMAGE_IS_EDIT, mOpTimes > 0);
+
+            FileUtil.ablumUpdate(this, saveFilePath);
+            setResult(RESULT_OK, shareIntent);
+            startActivity(shareIntent);
+        }
+        else
+            Toast.makeText(EditImageActivity.this,"Save Image first", Toast.LENGTH_SHORT).show();
```
Comment (pa1pal) [inline app/src/main/java/vn/mbm/phimp/me/editor/editimage/EditImageActivity.java]: avoid hardcoded strings. Use strings resource.

File: app/src/main/res/layout/activity_image_edit.xml [hunk 3]
```diff
@@ -21,12 +21,12 @@
             android:src="@drawable/back_arrow" />
 
         <TextView
-            android:id="@+id/title"
+            android:id="@+id/shareImage"
```
Comment (pa1pal) [inline app/src/main/res/layout/activity_image_edit.xml]: use underscores in ID name, not camel-case

File: app/src/main/java/vn/mbm/phimp/me/editor/editimage/EditImageActivity.java (hunk 1 above)
Comment (Subhankar29) [inline app/src/main/java/vn/mbm/phimp/me/editor/editimage/EditImageActivity.java]: Ok, I will change that.

File: app/src/main/java/vn/mbm/phimp/me/editor/editimage/EditImageActivity.java (hunk 2 above)
Comment (Subhankar29) [inline app/src/main/java/vn/mbm/phimp/me/editor/editimage/EditImageActivity.java]: Yes, sure.

File: app/src/main/res/layout/activity_image_edit.xml (hunk 3 above)
Comment (Subhankar29) [inline app/src/main/res/layout/activity_image_edit.xml]: Will improvise.

**Comment by Subhankar29** [comment]
@pa1pal For the point 1.) I think the current flow of the app, that is after taking a photo then the user will be asked to edit the image, is fine. The user can go to the share Image activity from edit screen with or without editing the photo. 
2.)Yes, we can redirect to the share Image activity directly from the gallery.

**Comment by mohitmanuja** [comment]
@Shubhakar29 Please improve the UI and also add toolbar color according to the theme. Currently, toolbar name is showing "Phimpme" should be related to sharing image. 
And you hard coded the icons in a drawable folder (only hdpi)  for sharing an image like Facebook, Drupal. You can use IconicImageView and use font awesome icons. (See the reference in SettingActivity) 
For example Facebook Icon- http://fontawesome.io/icon/facebook-official/ 
To use this icon add the code in app:iiv_icon="HERE". 
So for Facebook, it should be app:iiv_icon=" faw-facebook-official". (Add w before -).

**Comment by Subhankar29** [comment]
@mohitmanuja Sure I will change the toolbar and the fonts. Regarding the UI It will be nice if we discuss the UI so that I can make necessary changes.  it is just a mock up as in future we will be adding Drupal, Wordpress and Joomla.

**Comment by Subhankar29** [comment]
@mohitmanuja, as I can see icons in Font Awesome, lacks colour element. Will look for another alternative then.

**Comment by mohitmanuja** [comment]
Okay you can use other alternative or use a background color according to icon with font awesome.

**Comment by Subhankar29** [comment]
@mohitmanuja, @pa1pal @vinaysajjanapu @anantprsd5 please review it. I have updated it as suggested.

**Pull request merged**